src/controllers/QueryRunner.ts

```diff
@@ -181,31 +181,45 @@ export default class QueryRunner {
         this.eventEmitter.emit('complete');
     }
 
-    public handleBatchComplete(result: QueryExecuteBatchCompleteNotificationResult): void {
+    public handleBatchStart(result: QueryExecuteBatchNotificationParams): void {
         let batch = result.batchSummary;
+
+        // Recalculate the start and end lines, relative to the result line offset
         if (batch.selection) {
             batch.selection.startLine = batch.selection.startLine + this._resultLineOffset;
             batch.selection.endLine = batch.selection.endLine + this._resultLineOffset;
         }
-        this._batchSets.push(batch);
-        this.eventEmitter.emit('batch', batch);
+
+        // Store the batch
+        this._batchSets[batch.id] = batch;
+        this.eventEmitter.emit('batchStart', batch);
+    }
+
+    public handleBatchComplete(result: QueryExecuteBatchNotificationParams): void {
+        let batchGiven = result.batchSummary;
+        let batchLocal = this._batchSets[batchGiven.id];
+
+        // Store the info we don't get from the initial batch notification
+        batchLocal.hasError = batchGiven.hasError;
```

Comment:
The main reason for this is to ensure that we don't override the result set summaries for the batch. It's not really a huge issue, so I'll go ahead and make this change.